app/lib/api_authentication/by_access_token.rb

```diff
@@ -108,8 +108,8 @@ def authenticated_token
       @authenticated_token = domain_account.access_tokens.find_from_value(access_token) if access_token
     end
 
-    def enforce_access_token_permission
-      PermissionEnforcer.enforce(authenticated_token, &Proc.new)
+    def enforce_access_token_permission(&block)
```

Comment:
I am not sure the changes in this file are actually related to Ruby upgrade. I think it was just fixing some IDE style warnings.